conda/plugins/__init__.py

```diff
@@ -28,6 +28,24 @@ class CondaSubcommand(NamedTuple):
         Optional[int],  # return code
     ]
 
+    def __eq__(self, other):
+        return other.name == self.name if isinstance(other, type(self)) else NotImplemented
+
+    def __lt__(self, other):
+        return other.name < self.name if isinstance(other, type(self)) else NotImplemented
+
+    def __le__(self, other):
+        return other.name <= self.name if isinstance(other, type(self)) else NotImplemented
+
+    def __gt__(self, other):
+        return other.name > self.name if isinstance(other, type(self)) else NotImplemented
+
+    def __ge__(self, other):
+        return other.name >= self.name if isinstance(other, type(self)) else NotImplemented
+
+    def __hash__(self):
+        return hash(self.name)
```

Comment:
I wish we'd be able to use dataclasses or pydantic instead of overloading all of these :( @FFY00 Remind me, why did we go with NamedTuple again instead of data classes?

Reply:
Per my understanding of comparison magic methods:
- `cmp`/`__cmp__` was dropped in Python 3,
- the implied shortcut of only defining `__eq__` and one of `__lt__`, `__le__`, `__gt__`, or `__ge__` from Python 2 requires using [`@functools.total_ordering`](https://docs.python.org/3/library/functools.html#functools.total_ordering) in Python 3, the docs state that using this decorator has a performance hit so I opted not to use it